src/parse.rs

```diff
@@ -222,6 +223,26 @@ pub fn parse_mailbox(mut lines: &[u8]) -> Result<Mailbox> {
     }
 }
 
+pub fn parse_search_ids(lines: &[u8]) -> Result<Vec<u32>> {
```

Comment:
I wonder if this should return a `HashSet<u32>` instead. Thoughts?

Reply:
While the RFC does not mention anything, shouldn't ids be listed only once anyway? I didn't test it but my guess is that `Vec` should be more performant given that it does not have to bother hashing keys?

I guess the choice is: should we respect the server's response in the odd case that an id is mentioned multiple times?

Reply:
You're probably right that a `Vec` will be faster, but at the same time it is unlikely that an application is bottlenecked by IMAP :) The reason I propose `HashSet` is that it has an API that is nicer for working with sets of values than a `Vec`. You can easily and efficiently do things like `.contains`. But I'm happy accepting the PR with either.

Reply:
I'm fine with using `HashSet` and I like having a nicer API but then maybe `parse_capabilities` and `parse_names` should switch to using `HashSet`s?

Reply:
Yes, you're probably right! Let's do that in a different PR though.